book/integration-service.md

```diff
@@ -0,0 +1,271 @@
+# Integration Service
+
+
+## Overview
+
+The Integration Service is composed of a couple controllers that facilitate automated testing of content produced by the HACBS build pipelines. It is responsible for integration testing capabilities.
+
+To support this capability, it needs to do a couple of things.
+
+- When a build pipeline completes, it creates an Snapshot CR representing a new collection of components that should be tested together.
+- When it sees a new Snapshot CR (created either by itself or by a user), it coordinates deployment of the application for testing. It does this by creating a new “ephemeral” StoneSoup Environment CRs which should in turn trigger the GitOps Service to provision environment (and a sub-workspace in CPS)
+- When it sees the environment is ready, it triggers Tekton PipelineRuns supplied by the user that run application tests and validation against the app running in that ephemeral Environment.
+- When it sees that all of the test PipelineRuns that it created have passed, it finishes its work by triggering a deployment of the application to the lowest known “static” environment defined by the user, usually staging.
+- Finally, if any automatic ReleasePlans have been created by the user in the workspace, it will create a Release CR signalling intent to release the tested content, to be carried out by the release-service (separate threat model).
+
+
+### Goals
```

Comment:
These feel like they are trying to be user stories.
Can the pronouns (we/you) be changed to specific personas?

Reply:
It would indeed be helpful to replace "we" with more specific things like "a developer" or "the build system".

Reply:
After giving this some thought, I think we can replace "we" with "The Integration service" in this whole section. Most - if not all - items refer to multiple personas who are using the Integration service, but it's the service that provides these functionalities in the end.

Reply:
Done